src/common/ceph_time.cc

```diff
@@ -49,7 +49,18 @@ namespace ceph {
     coarse_real_clock::time_point coarse_real_clock::now(
       const CephContext* cct) noexcept {
       struct timespec ts;
-      clock_gettime(CLOCK_REALTIME_COARSE, &ts);
+#if defined(CLOCK_REALTIME_COARSE)
+        // Linux systems have _COARSE clocks.
+        clock_gettime(CLOCK_REALTIME_COARSE, &ts);
+#elif defined(CLOCK_REALTIME_FAST)
+        // BSD systems have _FAST clocks.
+        clock_gettime(CLOCK_REALTIME_FAST, &ts);
+#else
+        // And if we find neither, you may wish to consult your system's
+        // documentation.
+#warning Falling back to CLOCK_REALTIME, may be slow.
+        clock_gettime(CLOCK_REALTIME, &ts);
+#endif
```

Comment:
On 4-2-2016 17:15, Kefu Chai wrote:
> In src/common/ceph_time.cc
> <https://github.com/ceph/ceph/pull/7515#discussion_r51895280>:
> 
>> @@ -49,7 +49,18 @@ namespace ceph {
>>      coarse_real_clock::time_point coarse_real_clock::now(
>>        const CephContext* cct) noexcept {
>>        struct timespec ts;
>> -      clock_gettime(CLOCK_REALTIME_COARSE, &ts);
>> +#if defined(CLOCK_REALTIME_COARSE)
>> +        // Linux systems have _COARSE clocks.
>> +        clock_gettime(CLOCK_REALTIME_COARSE, &ts);
>> +#elif defined(CLOCK_REALTIME_FAST)
>> +        // BSD systems have _FAST clocks.
>> +        clock_gettime(CLOCK_REALTIME_FAST, &ts);
>> +#else
>> +        // And if we find neither, you may wish to consult your system's
>> +        // documentation.
>> +#warning Falling back to CLOCK_REALTIME, may be slow.
>> +        clock_gettime(CLOCK_REALTIME, &ts);
>> +#endif
> 
> @wjwithagen <https://github.com/wjwithagen> FYI, #7466
> <https://github.com/ceph/ceph/pull/7466> by @adamemerson
> <https://github.com/adamemerson> also addresses this issue.

I know, but I "need" it to actually make any progress in getting things
compiling....
If there is an other way of doing this, just let me know...

This is a thing to consider anyways since quite a few people are
creating fixes for FreeBSD compiling. And several of the outstanding
pulls are conflicting and/or fixing the smae problem

And I'm not sure on how this will be handled when importing any of the
pulls. I sort of expect to rebase and fix merges once any of the
FreeBSD/Darwin/other pulls gets accepted.

--WjW